File: 0001-two-sum/0001-two-sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  (1...nums.length).each do |i|
    (i...nums.length).each do |j|
      if nums[j] + nums[j - i] == target
        return [j - i, j]
      end
    end
  end
  [-1, -1]
end

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const pairs = nums.flatMap((num1, i) =>
    nums.map((num2, j) => ({ i, j, sum: num1 + num2 }))
    );

    const match = pairs.find(pair => pair.i < pair.j && pair.sum === target);

    return match ? [match.i, match.j] : [-1, -1];

};

